### PR TITLE
Add pseudolocalization tutorial

### PR DESCRIPTION
**Pull request by angad-k**
Tutorial PR for the [pseudolocalization feature I'm adding to Godot as part of GSoC](https://github.com/godotengine/godot/pull/49361).

### CODE REVIEW

File: tutorials/i18n/pseudolocalization.rst
Comment (Calinou) [inline tutorials/i18n/pseudolocalization.rst:58]:
```suggestion
- ``replace_with_accents``: Replaces all characters in the string with their accented 
   variants. *"The quick brown fox jumped over the lazy dog"* will be converted to 
   *"Ŧh̀é q́üíćḱ ḅŕôŵή f́ôx́ ǰüm̀ṕéd́ ôṽéŕ ŧh̀é łáźý d́ôǵ"* when this setting is enabled.
   This can be used to spot untranslated strings that won't have accents,
   but is also useful to check for missing glyphs in the font(s) used by the project.
- ``double_vowels``: Doubles all the vowels in the string. It is a good approximation
   to simulate expansion of text during localization.
   This can be used to check for text that would overflow its container
   (such as buttons).
- ``fake_bidi``: Fake bidirectional text (simulates right-to-left text). This is useful to
   simulate right-to-left writing systems to check for potential layout issues that would occur in languages using right-to-left scripts.
- ``override``: Replaces all the characters in the string with an asterisk (``*``). This is useful for 
   quickly finding text that isn't being localized.
- ``expansion_ratio``: Can be used in cases where doubling the vowels isn't a sufficient 
   approximation. This setting pads the string with underscores (``_``) and expands it by the given ratio.
   An expansion ratio of ``0.3`` is sufficient for most practical cases; it will increase 
   the length of the string by 30%.
- ``prefix`` and ``suffix``: These properties can be used to specify a prefix and suffix to wrap
   the text in.
- ``skip_placeholders``: Skips placeholders for string formatting like ``%s`` and ``%f``. 
   This is useful to identify places where more arguments are required for the formatted
   string to display correctly.
```

File: tutorials/i18n/pseudolocalization.rst
Comment (Calinou) [inline tutorials/i18n/pseudolocalization.rst:82]: Use shorthand syntax for code blocks, and only add 1 newline at the end of file:

```suggestion
The following code snippet shall turn on ``replace_with_accents`` and ``double_vowels`` properties
and then call ``reload_pseudolocalization()`` for the changes to get reflected::

    ProjectSettings.set("internationalization/pseudolocalization/replace_with_accents", true)
    ProjectSettings.set("internationalization/pseudolocalization/double_vowels", true)
	TranslationServer.reload_pseudolocalization()

```

**Review by Calinou — CHANGES_REQUESTED** [review]

File: tutorials/i18n/pseudolocalization.rst
Comment (angad-k) [reply tutorials/i18n/pseudolocalization.rst]: Should I also submit the demo on the asset library?

**Comment by mhilbrunner** [comment]
Looks good! :)

There is a build error with a wrong label to fix, and also the indentation at the very last line seems incorrect.

**Comment by mhilbrunner** [comment]
Now that this has landed with https://github.com/godotengine/godot/pull/51395, whats the state of this docs PR? :)

**Comment by angad-k** [comment]
Other than the failing check, this is pretty much done. Any idea why the check is failing? The label is correct according to me. [This](https://github.com/godotengine/godot/pull/51395/files#diff-9d8def4d07e1efed5044f7a0977be7fd25014fa52cfa044b209375f404a4f149R1570) is the property that the label's referring to.

**Comment by mhilbrunner** [comment]
Probably because the class docs weren't synced for master for some time, so the new properties/methods aren't available yet. We'll have to wait for Akien to get back from vacation and sync it eventually :)

**Comment by pycbouh** [comment]
Let it be a draft until we can actually consider merging it.

**Comment by YuriSizov** [comment]
This probably needs a force push to correctly rerun the CI

**Comment by angad-k** [comment]
Force push's done, ig it shall require an approval for the workflow to rerun from a maintainer.

**Review by skyace65 — APPROVED** [review]
Looks good to me!

**Comment by skyace65** [comment]
@Calinou @mhilbrunner would you like to review this again or is it good to go?

File: tutorials/i18n/pseudolocalization.rst
Comment (Calinou) [inline tutorials/i18n/pseudolocalization.rst:28]:
```suggestion
Enabling pseudolocalization and the configurations related to it is as simple as
toggling a checkbox in the project settings. These settings can be found in
**Project → Project Settings → General → Internationalization → Pseudolocalization**
after enabling the **Advanced** toggle in the project settings dialog:
```

File: tutorials/i18n/pseudolocalization.rst
Comment (Calinou) [inline tutorials/i18n/pseudolocalization.rst:68]:
```suggestion
Configuring pseudolocalization at run-time
------------------------------------------
```

File: tutorials/i18n/pseudolocalization.rst
Comment (Calinou) [inline tutorials/i18n/pseudolocalization.rst:33]:
```suggestion
Pseudolocalization can also be :ref:`toggled at runtime from a script <doc_pseudolocalization_runtime>`.
```

File: tutorials/i18n/pseudolocalization.rst
Comment (Calinou) [inline tutorials/i18n/pseudolocalization.rst:73]:
```suggestion
Pseudolocalization can be toggled at runtime using the
:ref:`pseudolocalization_enabled<class_TranslationServer_property_pseudolocalization_enabled>` property
in TranslationServer.
However, if runtime configuration of pseudolocalization properties is required, 
```

File: tutorials/i18n/pseudolocalization.rst
Comment (Calinou) [inline tutorials/i18n/pseudolocalization.rst:30]: Use WebP screenshot (see the "changes requested" review):

```suggestion
.. image:: img/pseudolocalization_settings.webp
```

**Review by Calinou — CHANGES_REQUESTED** [review]
Please replace the screenshot with this WebP image (and apply the suggestion to change the path in the documentation accordingly):

[pseudolocalization_settings.zip](https://github.com/godotengine/godot-docs/files/10508594/pseudolocalization_settings.zip)

This image has better compression and has the superfluous "Display" tooltip removed.

**Comment by angad-k** [comment]
Hey, made the requested changes - sorry for the massive delay- had looked at the review and then it slipped out of my mind until now. 😅

File: tutorials/i18n/pseudolocalization.rst
Comment (Calinou) [inline tutorials/i18n/pseudolocalization.rst:77]:
```suggestion
:ref:`ProjectSettings.set_setting(property, value) <class_ProjectSettings_method_set_setting>`
```

File: tutorials/i18n/pseudolocalization.rst
Comment (Calinou) [inline tutorials/i18n/pseudolocalization.rst:85]:
```suggestion
    ProjectSettings.set_setting("internationalization/pseudolocalization/replace_with_accents", true)
    ProjectSettings.set_setting("internationalization/pseudolocalization/double_vowels", true)
```

File: tutorials/i18n/pseudolocalization.rst
Comment (Calinou) [inline tutorials/i18n/pseudolocalization.rst:63]:
```suggestion
- ``replace_with_accents``: Replaces all characters in the string with their accented
  variants. *"The quick brown fox jumped over the lazy dog"* will be converted to
  *"Ŧh̀é q́üíćḱ ḅŕôŵή f́ôx́ ǰüm̀ṕéd́ ôṽéŕ ŧh̀é łáźý d́ôǵ"* when this setting is enabled.
  This can be used to spot untranslated strings that won't have accents,
  but is also useful to check for missing glyphs in the font(s) used by the project.
- ``double_vowels``: Doubles all the vowels in the string. It is a good approximation
  to simulate expansion of text during localization.
  This can be used to check for text that would overflow its container
  (such as buttons).
- ``fake_bidi``: Fake bidirectional text (simulates right-to-left text). This is useful to
  simulate right-to-left writing systems to check for potential layout issues that would occur in languages using right-to-left scripts.
- ``override``: Replaces all the characters in the string with an asterisk (``*``). This is useful for
  quickly finding text that isn't being localized.
- ``expansion_ratio``: Can be used in cases where doubling the vowels isn't a sufficient
  approximation. This setting pads the string with underscores (``_``) and expands it by the given ratio.
  An expansion ratio of ``0.3`` is sufficient for most practical cases; it will increase
  the length of the string by 30%.
- ``prefix`` and ``suffix``: These properties can be used to specify a prefix and suffix to wrap
  the text in.
- ``skip_placeholders``: Skips placeholders for string formatting like ``%s`` and ``%f``.
  This is useful to identify places where more arguments are required for the formatted
  string to display correctly.
```

**Review by Calinou — APPROVED** [review]
Thanks!

File: tutorials/i18n/pseudolocalization.rst
Comment (Calinou) [inline tutorials/i18n/pseudolocalization.rst:21]:
```suggestion
.. note:: The official demo for pseudolocalization can be used to play around with 
          the various properties and observe their effects. You can
          `download it from the godot-demo-projects GitHub repository <https://github.com/godotengine/godot-demo-projects/tree/4.0-dev/gui/pseudolocalization>`_.
```

**Pull request merged**
